### ISSUE TITLE
ImportError: cannot import name 'checkpoint' from 'transformers.models.t5.modeling_t5'

### ISSUE THREAD

**Issue by Dylan-LDR** [vimalabs/VIMA#47]
Thanks for your great work. I successfully install vima and vima_bench. However, when I try to run the example by '''python3 scripts/example.py --ckpt=../VimaBench/ckpts/200M.ckpt --partition=placement_generalization --task=follow_order''', I face the problem of failure to import checkpoint when import modeling_t5 at vima/nn/prompt_encoder/prompt_encoder.py. Did I miss some requirements and how can I fix this?

**Comment by Dylan-LDR** [comment]
I did check the source code and document of Transformers modeling_t5 but I could not find checkpoint definition.

**Comment by Zero-coder** [comment]
same question +1 @yunfanjiang

**Comment by Zero-coder** [comment]
Thanks for contributing such inspiring  work. I guess this problem might also related to HF version.

**Comment by Dylan-LDR** [comment]
Yes, I think the problem should lay in the version conflict of transformers. Maybe '''checkpoint''' is removed after a certain update so that I cannot find it in my installed transformers 4.36.1. Unfortunately I spot that there are no detailed versions of packages in requirement and it is hard to find supports from the community. Hope there could be more explanations and discussions.

**Comment by Abivishaq** [comment]
Yeah, the transformers version seems to be the issue. Version 4.20.0 worked for me.
```
pip install transformers==4.20.0 
```

**Comment by Zero-coder** [comment]
> Yes, I think the problem should lay in the version conflict of transformers. Maybe '''checkpoint''' is removed after a certain update so that I cannot find it in my installed transformers 4.36.1. Unfortunately I spot that there are no detailed versions of packages in requirement and it is hard to find supports from the community. Hope there could be more explanations and discussions. 

Thanks for your information, it's helpful.

**Comment by Zero-coder** [comment]
> Yeah, the transformers version seems to be the issue. Version 4.20.0 worked for me.
> 
> ```
> pip install transformers==4.20.0 
> ```

Thanks,  ` transformers==4.20.0 `, this version is helpful, however, then I encountered a new problem "[2023-12-19T02:58:39Z ERROR cached_path::cache] Max retries exceeded for https://huggingface.co/t5-base/resolve/main/tokenizer.json
Traceback (most recent call last):
  File "/home/jmw/VIMA-main/scripts/example.py", line 74, in <module>
    tokenizer = Tokenizer.from_pretrained("t5-base")
Exception: Model "t5-base" on the Hub doesn't have a tokenizer"

**Comment by Zero-coder** [comment]
Those are related solution, I am trying to solve.

https://github.com/vimalabs/VIMA/issues/20
https://github.com/vimalabs/VIMA/issues/42

**Comment by Dylan-LDR** [comment]
https://github.com/vimalabs/VIMA/issues/20#issuecomment-1801600549 
This reply works for me.

**Comment by Zero-coder** [comment]
> [#20 (comment)](https://github.com/vimalabs/VIMA/issues/20#issuecomment-1801600549) This reply works for me.

Hello, thanks for your information.
Can you give more details about this, I tried this, and I encounter with 
![4318f238e202e27c0e9e247cad36d5b](https://github.com/vimalabs/VIMA/assets/54145971/7407b296-e31b-4199-9279-e0ccf5ab0446)

**Comment by Zero-coder** [comment]
![024d101a3d1a8f3e60df064699098cb](https://github.com/vimalabs/VIMA/assets/54145971/6a7754d1-3667-4800-b62e-be4450f8642e)
and this is my file structure.

**Comment by Zero-coder** [comment]
@Dylan-LDR

**Comment by Dylan-LDR** [comment]
> > [#20 (comment)](https://github.com/vimalabs/VIMA/issues/20#issuecomment-1801600549) This reply works for me.
> 
> Hello, thanks for your information. Can you give more details about this, I tried this, and I encounter with ![4318f238e202e27c0e9e247cad36d5b](https://private-user-images.githubusercontent.com/54145971/291521439-7407b296-e31b-4199-9279-e0ccf5ab0446.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI5NzU0NTAsIm5iZiI6MTcwMjk3NTE1MCwicGF0aCI6Ii81NDE0NTk3MS8yOTE1MjE0MzktNzQwN2IyOTYtZTMxYi00MTk5LTkyNzktZTBjY2Y1YWIwNDQ2LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE5VDA4MzkxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJlM2I4MDQxYmNhNzZkYjY0MWMwNTQwMDI1MzIwNDY5ZmUzODIzMzMwODNjM2EwYTU1ZmJlYjQyYmNmODVkODAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.cOnVs7ZApLYhecjVsMedqqC7K5KVqWB-_RxitZe0WD0)

I just follow the instruction of https://github.com/vimalabs/VIMA/issues/20#issuecomment-1801600549 and load the tokenizer from local directory. It works fine for me totally. I guess the assertion is caused by the miss of VIMA model checkpoint. Did you ever download the checkpoint file 200M.ckpt from HF and place it into your code directory? I cannot find it from the file structure you posted.

**Comment by Zero-coder** [comment]
> > > [#20 (comment)](https://github.com/vimalabs/VIMA/issues/20#issuecomment-1801600549) This reply works for me.
> > 
> > 
> > Hello, thanks for your information. Can you give more details about this, I tried this, and I encounter with ![4318f238e202e27c0e9e247cad36d5b](https://private-user-images.githubusercontent.com/54145971/291521439-7407b296-e31b-4199-9279-e0ccf5ab0446.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI5NzU0NTAsIm5iZiI6MTcwMjk3NTE1MCwicGF0aCI6Ii81NDE0NTk3MS8yOTE1MjE0MzktNzQwN2IyOTYtZTMxYi00MTk5LTkyNzktZTBjY2Y1YWIwNDQ2LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE5VDA4MzkxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJlM2I4MDQxYmNhNzZkYjY0MWMwNTQwMDI1MzIwNDY5ZmUzODIzMzMwODNjM2EwYTU1ZmJlYjQyYmNmODVkODAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.cOnVs7ZApLYhecjVsMedqqC7K5KVqWB-_RxitZe0WD0)
> 
> I just follow the instruction of [#20 (comment)](https://github.com/vimalabs/VIMA/issues/20#issuecomment-1801600549) and load the tokenizer from local directory. It works fine for me totally. I guess the assertion is caused by the miss of VIMA model checkpoint. Did you ever download the checkpoint file 200M.ckpt from HF and place it into your code directory? I cannot find it from the file structure you posted.

Many thanks, it's working after following your suggestion. 🙂🌱

**Comment by jpbalarini** [comment]
Downgrading to version `pip install transformers==4.34.1` did the trick for me.
